### PR TITLE
`ModuleId` to `PalletId` - part of #8372

### PR DESCRIPTION
**Pull request by AurevoirXavier**
Related #8372

Polkadot companion: https://github.com/paritytech/polkadot/pull/2835

@thiolliere

### CODE REVIEW

**Review by kianenigma — COMMENTED** [review]
I think we shouldn't put `PalletId` in `sp-`.

File: primitives/runtime/src/lib.rs [hunk 1]
```diff
@@ -160,9 +160,9 @@ use traits::{Verify, Lazy};
 
 /// A module identifier. These are per module and should be stored in a registry somewhere.
```
Comment (thiolliere) [inline primitives/runtime/src/lib.rs:161]:
```suggestion
/// A pallet identifier. These are per pallet and should be stored in a registry somewhere.
```

**Review by thiolliere — APPROVED** [review]

**Comment by thiolliere** [comment]
> I think we shouldn't put `PalletId` in `sp-`.

Maybe it should be moved to frame-support, I don't know why it is in primitives considering client doesn't need it.

**Comment by bkchr** [comment]
Yeah @AurevoirXavier please move `PalletId` to `frame-support`.

**Comment by AurevoirXavier** [comment]
> Yeah @AurevoirXavier please move `PalletId` to `frame-support`.

Where should I place it? Currently I place it on the frame-support's root (lib.rs)

**Comment by thiolliere** [comment]
root is ok to me, but compilation needs to be fixed

**Comment by AurevoirXavier** [comment]
> root is ok to me, but compilation needs to be fixed

Okay. Companion is on the way.

**Comment by AurevoirXavier** [comment]
> root is ok to me, but compilation needs to be fixed

On the way.

Also a companion for polkadot.

**Review by bkchr — APPROVED** [review]

**Comment by AurevoirXavier** [comment]
@thiolliere I also change the subcommand `moduleid` to `palletid`

**Review by bkchr — APPROVED** [review]

**Comment by bkchr** [comment]
@AurevoirXavier this requires a polkadot companion.

**Comment by AurevoirXavier** [comment]
> @AurevoirXavier this requires a polkadot companion.

Done

**Review by shawntabrizi — APPROVED** [review]

**Comment by shawntabrizi** [comment]
bot merge

**Pull request merged**